lua/neo-tree/ui/inputs.lua

```diff
@@ -17,6 +17,13 @@ M.show_input = function(input, callback)
   if config.enable_normal_mode_for_inputs and input.prompt_type ~= "confirm" then
```

Comment:
I think what we can do is to change this to:

```lua
if input.prompt_type ~= "confirm" then
  vim.schedule(function()
    if config.enable_normal_mode_for_inputs then
      vim.cmd("stopinsert")
    end
    events.fire_event(events.NEO_TREE_POPUP_INPUT_READY)
  end)
end
```

Then people can use the NEO_TREE_POPUP_INPUT_READY event for custom actions without altering any existing configuration or introducing a hack we don't even want to document.

Reply:
If we add this, what's the point of enable_normal_mode option now?

Reply:
Backwards compatibility. I would be in favor of deprecating it.

Reply:
Okay. I'll fix that tomorrow.